### PR TITLE
Add Comparator and DAC

### PR DESCRIPTION
**Pull request by usbalbin**
Add modified src/dac.rs and src/comparator.rs from g0xx-hal. Also added support for internal connection between DAC and comparator.

Tested on nucleo-g474re (stm32g474re)

### CODE REVIEW

File: examples/comp_w_dac.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#[entry]
+fn main() -> ! {
+    let dp = stm32::Peripherals::take().expect("cannot take peripherals");
+    let cp = cortex_m::Peripherals::take().expect("cannot take core peripherals");
+
+    let mut rcc = dp.RCC.constrain();
+    let mut delay = cp.SYST.delay(&rcc.clocks);
+
+    let gpioa = dp.GPIOA.split(&mut rcc);
+    let dac1ch1 = dp.DAC1.constrain((gpioa.pa4, Dac1IntSig1), &mut rcc);
```
Comment (usbalbin) [inline examples/comp_w_dac.rs:28]: Not sure how this should best be done.. There is no way for the user to know whether to specify Dac1IntSig1 or Dac1IntSig2(besides a not too nice compile error) in order to create a dac in with a mode compatible with the comparator.

Suggestions welcome :)

File: src/comparator.rs [hunk 2]
```diff
@@ -0,0 +1,826 @@
...
+        self
+    }*/
+
+    /*
+    /// Sets the output to be Comparator 1 XOR Comparator 2.
+    /// Used to implement window comparator mode.
+    pub fn output_xor(mut self) -> Self {
+        self.output_xor = true;
+        self
+    }*/
```
Comment (usbalbin) [inline src/comparator.rs:155]: AFAIK this is not supported by stm32g4 processors. Should this be removed?

File: src/comparator.rs [hunk 3]
```diff
@@ -0,0 +1,826 @@
...
+    };
+}
+
+impl_comparator!(COMP1, comp1, ExtiEvent::COMP1);
+impl_comparator!(COMP2, comp2, ExtiEvent::COMP2);
+
+/*
+/// Uses two comparators to implement a window comparator.
+/// See Figure 69 in RM0444 Rev 5.
+pub struct WindowComparator<U, L, ED> {
```
Comment (usbalbin) [inline src/comparator.rs:563]: Same thing with window mode, not supported

File: src/dac.rs [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+                /*pub fn enable_unbuffered(self) -> $CX<MODE_BITS, EnabledUnbuffered> {
+                    let dac = unsafe { &(*<$DAC>::ptr()) };
+
+                    dac.dac_mcr.modify(|_, w| unsafe { w.$mode().bits(2) });
+                    dac.dac_cr.modify(|_, w| w.$en().set_bit());
+
+                    $CX {
+                        _enabled: PhantomData,
+                    }
+                }*/
```
Comment (usbalbin) [inline src/dac.rs:203]: Buffered vs unbuffered is pretty much chosen by MODE_BITS. Not sure how we want to handle that

**Comment by no111u3** [comment]
@usbalbin please update the PR if it still needed.

File: src/opamp.rs [hunk 5]
```diff
@@ -45,6 +45,8 @@
 //! }
 //! ```
 
+// TODO: Add support for locking using the `LOCK` bit in `OPAMPx_CSR`
```
Comment (no111u3) [inline src/opamp.rs:48]: Needs to update to current main branch

File: src/dac.rs (hunk 4 above)
Comment (no111u3) [reply src/dac.rs:203]: Use input argument as selector between modes

File: src/comparator.rs (hunk 2 above)
Comment (no111u3) [reply src/comparator.rs:162]: Yes we should. We don't need to store code in comments.

File: src/comparator.rs (hunk 3 above)
Comment (no111u3) [reply src/comparator.rs:596]: Also remove every inapplicable code.

File: src/comparator.rs [hunk 6]
```diff
@@ -0,0 +1,865 @@
...
+        }
+    };
+    ($($COMP:ident: $pin:ident, $AF:ident,)+) => {$(
+        output_pin!($COMP, $pin, $AF, PushPull, into_alternate);
+        output_pin!($COMP, $pin, $AF, OpenDrain, into_alternate_open_drain);
+    )+};
+}
+
+// TODO: look up alternate functions for more devices than g474
+// https://www.mouser.se/datasheet/2/389/stm32g474cb-1600828.pdf#page=73
```
Comment (no111u3) [inline src/comparator.rs:825]: please use the documents from ST site only. also use their public id like as ANxxx, RMxxx, etc.

File: src/comparator.rs [hunk 7]
```diff
@@ -0,0 +1,865 @@
...
+
+    /// Does this input rely on dividing Vrefint using an internal resistor divider
+    ///
+    /// This is only relevant for `RefintInput` other than `RefintInput::VRefint`
+    fn use_resistor_divider(&self) -> bool;
+
+    fn setup(&self, comp: &C);
+}
+
+/*
```
Comment (no111u3) [inline src/comparator.rs:203]: Please remove the unused or inapplicable code.

**Review by no111u3 — CHANGES_REQUESTED** [review]

**Comment by usbalbin** [comment]
I believe I have resolved your comments. Btw thanks a lot :)

**Review by no111u3 — APPROVED** [review]

**Comment by no111u3** [comment]
@usbalbin thank you so much. I've to merge it.

**Pull request merged**